src/conduits/download-rsc-conduit.ts

```diff
@@ -0,0 +1,97 @@
+import fs from 'fs-extra';
+import {DlpDBInfoType, DlpOpenConduitReqType} from '../protocols/dlp-commands';
+import {DlpConnection} from '../protocols/sync-connections';
+import {DATABASES_STORAGE_DIR, SyncType} from '../sync-utils/sync-device';
+import {ConduitData, ConduitInterface} from './conduit-interface';
+import {RawPdbDatabase, RawPdbRecord, RecordEntryType} from 'palm-pdb';
+import {
+  writeRawDbToFile,
+  ReadDbOptions,
+  readRawDb,
+} from '../sync-utils/read-db';
+import debug from 'debug';
+
+const log = debug('palm-sync').extend('conduit').extend('download-new');
+
+/**
+ * This conduit download resources that exists on the Palm, but not on PC.
+ */
+export class DownloadNewResourcesConduit implements ConduitInterface {
+  getName(): String {
+    return 'download new resources from Palm';
+  }
+  async execute(
+    dlpConnection: DlpConnection,
+    conduitData: ConduitData
+  ): Promise<void> {
+    if (conduitData.dbList == null) {
+      throw new Error('dbList is mandatory for this Conduit');
+    }
+
+    let downloadCount = 0;
+
+    await dlpConnection.execute(DlpOpenConduitReqType.with({}));
+
+    for (let index = 0; index < conduitData.dbList.length; index++) {
+      const dbInfo = conduitData.dbList[index];
+
+      const ext = dbInfo.dbFlags.resDB ? 'prc' : 'pdb';
+      const fileName = `${dbInfo.name}.${ext}`;
+
+      const resourceExists = await fs.exists(
+        `${conduitData.palmDir}/${DATABASES_STORAGE_DIR}/${fileName}`
+      );
+
+      if (!resourceExists) {
+        try {
+          log(
```

Comment:
Yup, that's exactly what I mentioned that could be clean up. The major difference here from the mentioned functions is that we must remove the dirty and busy flags, otherwise in the next sync the files will be improperly sync'd. But I did implement that later and haven't updated here haha Done!
![image](https://github.com/jichu4n/palm-sync/assets/4419930/38971870-8f60-4ccd-abdd-e89790f62bb7)